### PR TITLE
[Merged by Bors] - Implement var initializers in for-in loops

### PR DESCRIPTION
**Pull request by jedel1043**
This Pull Request implements [Initializers in ForIn Statement Heads](https://tc39.es/ecma262/#sec-initializers-in-forin-statement-heads) from the Annex B. This also cleans up the "annex-b" feature to be able to disable it with `--no-default-features`, since I couldn't test the error messages when the feature is disabled.

### CODE REVIEW

File: boa_parser/src/parser/statement/labelled_stm/mod.rs [hunk 1]
```diff
@@ -65,18 +65,25 @@ where
...
+                if cfg!(not(feature = "annex-b")) || strict =>
+            {
                 return Err(Error::general(
-                    "In strict mode code, functions can only be declared at top level or inside a block.",
-                    next_token.span().start()
+                    format!(
+                        "{}functions can only be declared at top level or inside a block.",
+                        if strict { "in strict mode code, " } else { "" },
+                    ),
+                    next_token.span().start(),
```
Comment (raskad) [inline boa_parser/src/parser/statement/labelled_stm/mod.rs:76]: Can this not be `function_declaration_in_if()`?

**Review by raskad — APPROVED** [review]

File: boa_parser/src/parser/statement/labelled_stm/mod.rs (hunk 1 above)
Comment (jedel1043) [reply boa_parser/src/parser/statement/labelled_stm/mod.rs]: Yep :)

**Review by HalidOdat — APPROVED** [review]
Looks good to me! :)

**Pull request closed without merging**